dask/dataframe/tests/test_reshape.py

```diff
@@ -191,6 +191,32 @@ def test_pivot_table(values, aggfunc):
     assert_eq(res, exp)
 
 
+@pytest.mark.parametrize("values", ["B", ["D"], ["B", "D"]])
+@pytest.mark.parametrize("aggfunc", ["first", "last"])
+def test_pivot_table_firstlast(values, aggfunc):
```

Comment:
Yes, my thought was to test this functionality for non-numeric data, i.e. strings in column "D" of the dataframe. Perhaps those tests should be separate from the pure 'first' and 'last' tests? Or maybe working with non-numeric data needs to be solved at a more fundamental level? Initially I tried using the existing test function, but as the additional column contained non-numeric data, that did not work.

Reply:
I agree. I have removed my test_pivot_table_firstlast function and only use the existing test_pivot_table function.

Reply:
Ah, yeah, that metadata generation step will need to change since`first`/`last` return non-numeric data (sorry, I missed that before during review). Testing with `assert_eq` is important, since it checks that the metadata (column names, dtypes, etc...) on the dask collections matches their computed values - if that test fails it's a sign that the metadata isn't being properly computed.

You'll need to adjust the line computing `meta` to make it properly infer dtypes for `first`/`last`.

Reply:
Yes, that is much better. Datatypes are now inferred for first and last.